Sources/StreamChatUI/CommonViews/ContainerStackView.swift

```diff
@@ -4,70 +4,81 @@
 
 import UIKit
 
-extension UIView {
-    func flexible(axis: NSLayoutConstraint.Axis) -> Self {
-        setContentHuggingPriority(.lowest, for: axis)
-        return self
-    }
-}
+extension ContainerStackView {
 
-extension NSLayoutConstraint {
-    func priority(_ p: UILayoutPriority) -> Self {
-        priority = p
-        return self
-    }
+    /// Describes the size distribution of the arranged subviews in a container stack view.
+    public struct Distribution: Equatable {
+        /// Makes the arranged subviews with their natural size.
+        public static let natural = Distribution(rawValue: 0)
+        /// Makes the arranged subviews all with the same size.
+        public static let equal = Distribution(rawValue: 1)
 
-    func priority(_ p: Float) -> Self {
-        priority = UILayoutPriority(p)
-        return self
+        private let rawValue: Int
     }
-}
 
-extension UILayoutPriority {
-    static let lowest = UILayoutPriority(defaultLow.rawValue / 2.0)
-}
+    /// Describes the alignment of the arranged subviews in perpendicular to the container's axis.
+    public struct Alignment: Equatable {
+        /// Makes the arranged subviews so that they **fill** the available space perpendicular to the container’s axis.
+        public static let fill = Alignment(rawValue: 0)
+        /// Makes the arranged subviews align to the **leading edge** in a **vertical axis** container.
+        /// Makes the arranged subviews align to the **top edge** in a **horizontal axis** container.
+        public static let axisLeading = Alignment(rawValue: 1)
+        /// Makes the arranged subviews align to the **trailing edge** in a **vertical axis** container.
+        /// Makes the arranged subviews align to the **bottom edge** in a **horizontal axis** container.
+        public static let axisTrailing = Alignment(rawValue: 2)
+        /// Makes the arranged subviews align to the **center** along its axis.
+        public static let center = Alignment(rawValue: 3)
 
-extension CGFloat {
-    public static let auto: CGFloat = .infinity
+        private let rawValue: Int
+    }
 }
 
+/// A view that works similar to a `UIStackView` but in a more simpler and flexible way.
+/// The aim of this view is to make UI customizability easier in the SDK.
 public class ContainerStackView: UIView {
+
+    private var customConstraints: [NSLayoutConstraint] = []
+
+    private var customSpacingByView: [UIView: CGFloat] = [:]
+
+    private var spacingConstraintsByView: [UIView: NSLayoutConstraint] = [:]
+
+    private var hideConstraintsByView: [UIView: (width: NSLayoutConstraint, height: NSLayoutConstraint)] = [:]
```

Comment:
Shouldn't setting either `height=0` OR `width=0` work? 
```swift
private var hideConstraintsByView: [UIView: NSLayoutConstraint] = [:]

// hideArrangedSubview
hideConstraintsByView[subview] = {
 if axis == .horizontal {
  return subview.widthAnchor.constraint(equalToConstant: 0)
 } else {
  return subview.heightAnchor.constraint(equalToConstant: 0)
 }
}()
```

Reply:
Hum, maybe 🤔 But I was thinking right now... what Happens if the view that is being hidden has the bigger width in a vertical stack, and the bigger height in a horizontal stack? 🤔 Won't that stack the stack to have empty space?

Reply:
I think we accidentally introduced a regression when changing the container in the past. I think both of the constraints are needed because every view can affect both axes.

Vertical container view like this:
```
┌ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┐
 ┌───────────────────────────────────────────┐ 
││                                           ││
 │                     A                     │ 
││                                           ││
 └───────────────────────────────────────────┘ 
│┌──────────────────────┐                     │
 │                      │                      
││          B           │                     │
 │                      │                      
│└──────────────────────┘                     │
 ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ 
```
should look like this when you hide the A. Even though the axis is vertical, the A affected the size of the container in both axes.

```           
┌ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─                      
 ┌──────────────────────┐│                     
││                      │                      
 │          B           ││                     
││                      │                      
 └──────────────────────┘│                     
└ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─                      
```

Reply:
✅  Done! The new hiding algorithm is implemented